StyleCop.Analyzers/StyleCop.Analyzers/Helpers/XmlCommentHelper.cs

```diff
@@ -268,22 +267,109 @@ internal static string GetText(XmlTextSyntax textElement, bool normalizeWhitespa
                 return null;
             }
 
-            StringBuilder stringBuilder = StringBuilderPool.Allocate();
+            string result = string.Empty;
+
+            StringBuilder stringBuilder = null;
 
             foreach (var item in textElement.TextTokens)
             {
-                stringBuilder.Append(item);
+                if (result.Length == 0)
+                {
+                    result = item.ToString();
+                }
+                else
+                {
+                    if (stringBuilder == null)
+                    {
+                        stringBuilder = StringBuilderPool.Allocate();
+                        stringBuilder.Append(result);
+                    }
+
+                    stringBuilder.Append(item.ToString());
+                }
+            }
+
+            if (stringBuilder != null)
+            {
+                result = StringBuilderPool.ReturnAndFree(stringBuilder);
             }
 
-            string result = StringBuilderPool.ReturnAndFree(stringBuilder);
             if (normalizeWhitespace)
             {
-                result = Regex.Replace(result, @"\s+", " ");
+                result = result.NormalizeWhiteSpace();
```

Comment:
just curious, @stephentoub on the latest .NET (of course this is broader targeted) would you expect regex replace using pre generated regex to have comparable performance to open coded?

I didn't measure, but fwiw 
```c#
        [GeneratedRegex(@"[\s^ ]+")]
        private static partial Regex GetRegex();
```
generates code in part like
```c#
			private bool TryFindNextPossibleStartingPosition(ReadOnlySpan<char> inputSpan)
			{
				int pos = runtextpos;
				if ((uint)pos < (uint)inputSpan.Length)
				{
					ReadOnlySpan<char> span = inputSpan.Slice(pos);
					for (int i = 0; i < span.Length; i++)
					{
						char ch;
						if (((ch = span[i]) < '\u0080') ? ((byte)("㸀\0\u0001\0\0䀀\0\0"[(int)ch >> 4] & (1 << (ch & 0xF))) != 0) : RegexRunner.CharInClass(ch, "\0\u0004\u0001 !^_d"))
						{
							runtextpos = pos + i;
							return true;
						}
					}
				}
				runtextpos = inputSpan.Length;
				return false;
```
the code below hides some of this inside `char.IsWhiteSpace`

Reply:
and will return the same string if there are no non-space whitespace chars.

Reply:
After addressing the single line issue and regular expression issue, more optimizations could be having much less return.

Reply:
> just curious, @stephentoub on the latest .NET (of course this is broader targeted) would you expect regex replace using pre generated regex to have comparable performance to open coded?

Yes.

Note that in .NET 8, the generated code now looks like this:
```C#
                private bool TryFindNextPossibleStartingPosition(ReadOnlySpan<char> inputSpan)
                {
                    int pos = base.runtextpos;
                    
                    // Empty matches aren't possible.
                    if ((uint)pos < (uint)inputSpan.Length)
                    {
                        // The pattern begins with a whitespace character.
                        // Find the next occurrence. If it can't be found, there's no match.
                        int i = inputSpan.Slice(pos).IndexOfAnyWhiteSpace();
                        if (i >= 0)
                        {
                            base.runtextpos = pos + i;
                            return true;
                        }
                    }
                    
                    // No match found.
                    base.runtextpos = inputSpan.Length;
                    return false;
                }
```
where that IndexOfAnyWhitespace is emitted as:
```C#
        /// <summary>Finds the next index of any character that matches a whitespace character.</summary>
        [MethodImpl(MethodImplOptions.AggressiveInlining)]
        internal static int IndexOfAnyWhiteSpace(this ReadOnlySpan<char> span)
        {
            int i = span.IndexOfAnyExcept(Utilities.s_asciiExceptWhiteSpace);
            if ((uint)i < (uint)span.Length)
            {
                if (char.IsAscii(span[i]))
                {
                    return i;
                }
        
                do
                {
                    if (char.IsWhiteSpace(span[i]))
                    {
                        return i;
                    }
                    i++;
                }
                while ((uint)i < (uint)span.Length);
            }
        
            return -1;
        }
        
        /// <summary>Supports searching for characters in or not in "\0\u0001\u0002\u0003\u0004\u0005\u0006\a\b\u000e\u000f\u0010\u0011\u0012\u0013\u0014\u0015\u0016\u0017\u0018\u0019\u001a\u001b\u001c\u001d\u001e\u001f!\"#$%&amp;'()*+,-./0123456789:;&lt;=&gt;?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~\u007f".</summary>
        internal static readonly IndexOfAnyValues<char> s_asciiExceptWhiteSpace = IndexOfAnyValues.Create("\0\u0001\u0002\u0003\u0004\u0005\u0006\a\b\u000e\u000f\u0010\u0011\u0012\u0013\u0014\u0015\u0016\u0017\u0018\u0019\u001a\u001b\u001c\u001d\u001e\u001f!\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~\u007f");
```
such that the search for the whitespace is vectorized.  Then the matching logic for finding all the contiguous whitespace currently looks like:
```C#
                    // Match a whitespace character atomically at least once.
                    {
                        int iteration = 0;
                        while ((uint)iteration < (uint)slice.Length && char.IsWhiteSpace(slice[iteration]))
                        {
                            iteration++;
                        }
                        
                        if (iteration == 0)
                        {
                            return false; // The input didn't match.
                        }
                        
                        slice = slice.Slice(iteration);
                        pos += iteration;
                    }
```
(though we're flirting with a change that'll result in that loop also changing to similarly be vectorized with IndexOfAnyExcept).

Replace itself also gets better in .NET 8.